GDJS/Runtime/howler-sound-manager/howler-sound-manager.ts

```diff
@@ -1,104 +1,296 @@
+     */
+    play(): this {
+      if (this.isLoaded()) {
+        const newID = this._howl.play(this._id === null ? undefined : this._id);
+        this._id = newID;
+
+        // Manually handle the play event before we have an ID.
+        // Before loading, howler won't register events as without an ID we cannot set a listener.
+        // Once we have an ID, we can transfer control of the events to howler.
+        // We also need to call them once as Howler doesn't for the first play event.
+        this._onPlay.forEach((func) => {
+          // Transfer the event to howler now that we have an ID
+          this.on('play', func);
+          func(newID);
+        });
+        this._oncePlay.forEach((func) => func(newID));
+        this._onPlay = [];
+        this._oncePlay = [];
+      } else this._howl.once('load', () => this.play()); // Play only once the howl is fully loaded
+
+      return this;
     }
 
-    pause() {
-      this._paused = true;
-      this._stopped = false;
-      return super.pause();
+    /**
+     * Pauses playback of the sound, saving the seek of playback.
+     * @returns The current instance for chaining.
+     */
+    pause(): this {
+      if (typeof this._id !== 'undefined') this._howl.pause(this._id);
+      return this;
+    }
+
+    /**
+     * Stops playback of the sound, resetting seek to 0.
+     * @returns The current instance for chaining.
+     */
+    stop(): this {
+      if (typeof this._id !== 'undefined') this._howl.stop(this._id);
+      return this;
+    }
+
+    /**
+     * Check if the sound is currently playing.
+     */
+    playing(): boolean {
+      return (
+        (this._id !== null ? this._howl.playing(this._id) : false) ||
+        !this.isLoaded() // Loading is considered playing
+      );
+    }
+
+    /**
+     * Check if the sound is currently paused.
+     */
+    paused(): boolean {
+      return !this.playing();
+    }
+
+    /**
+     * Check if the sound is currently stopped.
+     */
+    stopped(): boolean {
+      return this.paused() && this.getSeek() === 0;
+    }
+
+    /**
+     * Get the sound playback rate.
+     */
+    getRate(): float {
+      if (typeof this._id === 'undefined') return 0;
+      return this._howl.rate(this._id);
+    }
+
+    /**
+     * Set the playback rate.
+     * @returns The current instance for chaining.
+     */
+    setRate(rate: float): this {
+      if (typeof this._id !== 'undefined') {
+        rate = gdjs.HowlerSoundManager.clampRate(rate);
+        this._howl.rate(rate, this._id);
+      }
+      return this;
+    }
+
+    /**
+     * Get if the sound is looping.
+     */
+    getLoop(): boolean {
+      if (this._id === undefined) return false;
+      return this._howl.loop(this._id);
+    }
+
+    /**
+     * Set if the sound is looping.
+     * @returns The current instance for chaining.
+     */
+    setLoop(loop: boolean): this {
+      if (this._id !== null) this._howl.loop(loop, this._id);
```

Comment:
You're comparing id to null here, sounds like a bug because the rest is using undefined?
I would recommend moving everything to null - undefined will make it hard to debug if something is really undefined or is undefined on purpose. Better use always null.